client/src/app/queries/dependencies.ts

```diff
@@ -32,8 +34,28 @@ export const useFetchDependencies = (
     onError: (error) => console.log("error, ", error),
     keepPreviousData: true,
   });
+
+  const result = useMemo(() => {
+    if (!data) {
+      return { data: [], total: 0, params };
+    }
+
+    const syntheticData: WithUiId<AnalysisDependency>[] = data.data.map(
+      (dep) => ({
+        ...dep,
+        _ui_unique_id: `${dep.name}/${dep.provider}`,
+      })
+    );
+
+    return {
+      data: syntheticData,
+      total: data.total,
+      params: data.params,
+    };
+  }, [data, params]);
```

Comment:
I wonder if we can abstract out something like:

```ts
useWithUiId((dep) => `${dep.name}/${dep.provider}`);
```

This indicates to me that maybe internally the table-controls/batteries/helpers could just take a `getItemId` callback in the first place if you don't have a usable `idProperty`. Will factor that into thinking on current refactors over there.

LGTM as-is though if you want to merge.

Reply:
`idProperty: string | (item: TItem) -> string`?

;-)